pkg/optimizely/cache.go

```diff
@@ -95,7 +111,16 @@ func (c *OptlyCache) GetClient(sdkKey string) (*OptlyClient, error) {
 
 	// If we didn't "set" the key in this method execution then it was set in another thread.
 	// Recursively lookuping up the SDK key "should" only happen once.
-	return c.GetClient(sdkKey)
+	return c.GetClient(sdkKey, datafileAccessToken)
+}
+
+// UpdateConfigs is used to update config for all clients corresponding to a particular SDK key.
+func (c *OptlyCache) UpdateConfigs(sdkKey string) {
+	for clientInfo := range c.optlyMap.IterBuffered() {
+		if strings.HasPrefix(clientInfo.Key, sdkKey) {
+			clientInfo.Val.(*OptlyClient).UpdateConfig()
```

Comment:
Maybe to be safe, it could check the result of type cast attempt, log an error and return when it's unexpectedly not an `*OptlyClient`.